aws/resource_aws_glue_catalog_database_test.go

```diff
@@ -33,6 +34,28 @@ func TestAccAWSGlueCatalogDatabase_importBasic(t *testing.T) {
 	})
 }
 
+func TestAccAWSGlueCatalogDatabase_validation(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGlueDatabaseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccGlueCatalogDatabase_validation(acctest.RandString(256)),
+				ExpectError: regexp.MustCompile(`"name" cannot be more than 255 characters long`),
+			},
+			{
+				Config:      testAccGlueCatalogDatabase_validation(""),
+				ExpectError: regexp.MustCompile(`"name" must be at least 1 character long`),
+			},
+			{
+				Config:      testAccGlueCatalogDatabase_validation("ABCDEFG"),
+				ExpectError: regexp.MustCompile(`"name" can only contain lowercase alphanumeric characters`),
+			},
```

Comment:
You might want to add a test saying that the name is actually valid?

Reply:
Will do, I'm also adding additional commits for the table name since the same auto-lowercase process happens AWS side.